File: apps/research-front/lib/src/driver/api/json/article.dart
import 'package:json_annotation/json_annotation.dart';

part 'article.g.dart';

@JsonSerializable()
class ArticlesJson {
  final List<ArticleJson> articles;

  ArticlesJson(this.articles);

  factory ArticlesJson.fromJson(Map<String, dynamic> json) =>
    _$ArticlesJsonFromJson(json);
}

@JsonSerializable()
class ArticleJson {
  final String articleId;
  final ArticleOverviewJson overview;
  final Map<String, dynamic> content;

  ArticleJson(this.articleId, this.overview, this.content);
  factory ArticleJson.fromJson(Map<String, dynamic> json) => _$ArticleJsonFromJson(json);
}

@JsonSerializable()
class ArticleOverviewJson {
  final EditorJson editor;
  final String articleName;
  final String lastModified;
  final String thumbnail;
  final String description;

  ArticleOverviewJson(this.editor, this.articleName, this.lastModified, this.thumbnail, this.description);
  factory ArticleOverviewJson.fromJson(Map<String, dynamic> json) => _$ArticleOverviewJsonFromJson(json);
}

@JsonSerializable()
class EditorJson {
  final String editorName;
  final String editorIcon;

  EditorJson(this.editorName, this.editorIcon);
  factory EditorJson.fromJson(Map<String, dynamic> json) => _$EditorJsonFromJson(json);
}

File: apps/research-api/infrastructure/database/model/article.go
package model

import "time"

type Article struct {
	ArticleId    string    `gorm:"column:id"`
	Title        string    `gorm:"column:title"`
	Description  string    `gorm:"column:description"`
	Editor       string    `gorm:"column:editor"`
	Thumbnail    string    `gorm:"column:thumbnail"`
	LastModified time.Time `gorm:"column:last_modified"`
}

func (a Article) TableName() string {
	return "research.article"
}
